### PR TITLE
[AWS SageMaker] Add integration test for sample pipeline train

### PR DESCRIPTION
**Pull request by akartsky**
Add integration test for [sample pipeline train](https://github.com/kubeflow/pipelines/tree/master/samples/contrib/aws-samples/simple_train_pipeline)

### CODE REVIEW

File: components/aws/sagemaker/codebuild/integration-test.buildspec.yml [hunk 1]
```diff
@@ -7,18 +7,17 @@ env:
 phases:
   build:
     commands:
-      - cd components/aws
-      - docker build . -f ./sagemaker/tests/integration_tests/Dockerfile -t amazon/integration-test-image --quiet
+      - docker build . -f ./components/aws/sagemaker/tests/integration_tests/Dockerfile -t amazon/integration-test-image --quiet
```
Comment (gautamkmr) [inline components/aws/sagemaker/codebuild/integration-test.buildspec.yml:10]: what has changed here ?

File: components/aws/sagemaker/tests/integration_tests/sample_pipeline_tests/test_sample_train_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import pytest
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/sample_pipeline_tests/test_sample_train_pipeline.py:1]: merge this with training component test ?,

File: components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      DataSource:
+        S3DataSource:
+          S3Uri: s3://((DATA_BUCKET))/mnist_kmeans_example/train_data
+          S3DataType: S3Prefix
+          S3DataDistributionType: FullyReplicated
+      CompressionType: None
+      RecordWrapperType: None
+      InputMode: File
+  model_artifact_path: s3://((DATA_BUCKET))/mnist_kmeans_example/output
+  role: ((ROLE_ARN))
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml:19]: add test for spot as discussed ?

File: components/aws/sagemaker/tests/integration_tests/Dockerfile [hunk 4]
```diff
@@ -38,6 +38,7 @@ ENV S3_DATA_BUCKET="kfp-test-data"
 ENV MINIO_LOCAL_PORT=9000
 ENV KFP_NAMESPACE="kubeflow"
 
-COPY ./sagemaker/ .
+RUN mkdir kfp_repo
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/Dockerfile:41]: lets name it pipelines -> same as repository?

File: components/aws/sagemaker/codebuild/integration-test.buildspec.yml (hunk 1 above)
Comment (surajkota) [reply components/aws/sagemaker/codebuild/integration-test.buildspec.yml:10]: context of docker build changed to repo's root directory

Comment (akartsky) [reply components/aws/sagemaker/codebuild/integration-test.buildspec.yml:10]: build context of docker is different 

it used to be "components/aws"
now it is the root of the git repo

File: components/aws/sagemaker/tests/integration_tests/sample_pipeline_tests/test_sample_train_pipeline.py (hunk 2 above)
Comment (akartsky) [reply components/aws/sagemaker/tests/integration_tests/sample_pipeline_tests/test_sample_train_pipeline.py]: done

File: components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml (hunk 3 above)
Comment (akartsky) [reply components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml:22]: done

File: components/aws/sagemaker/tests/integration_tests/Dockerfile (hunk 4 above)
Comment (akartsky) [reply components/aws/sagemaker/tests/integration_tests/Dockerfile]: done

File: components/aws/sagemaker/tests/integration_tests/component_tests/test_train_component.py [hunk 5]
```diff
@@ -11,9 +11,21 @@
...
+            "resources/config/fsx-mnist-training",
+            marks=pytest.mark.fsx_test
+        ),
+        pytest.param(
+            "resources/config/sample-pipeline-training",
+            marks=pytest.mark.canary_test,
+        ),
+        pytest.param(
+            "resources/config/spot-sample-pipeline-training",
+            marks=pytest.mark.canary_test,
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/component_tests/test_train_component.py:27]: remove canary marker? and lets have only this test - spot-sample-pipeline-training. I think that would be sufficient ?

all other changes looks good

File: components/aws/sagemaker/tests/integration_tests/resources/config/spot-sample-pipeline-training/config.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        S3DataSource:
+          S3Uri: s3://((DATA_BUCKET))/mnist_kmeans_example/train_data
+          S3DataType: S3Prefix
+          S3DataDistributionType: FullyReplicated
+      CompressionType: None
+      RecordWrapperType: None
+      InputMode: File
+  spot_instance: "True"
+  checkpoint_config:
+    S3Uri: s3://((DATA_BUCKET))/train-checkpoints
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/resources/config/spot-sample-pipeline-training/config.yaml:23]: nit: lets keep this inside mnist_kmeans_example directory ?

File: components/aws/sagemaker/tests/integration_tests/component_tests/test_train_component.py (hunk 5 above)
Comment (akartsky) [reply components/aws/sagemaker/tests/integration_tests/component_tests/test_train_component.py]: done

File: components/aws/sagemaker/tests/integration_tests/resources/config/spot-sample-pipeline-training/config.yaml (hunk 6 above)
Comment (akartsky) [reply components/aws/sagemaker/tests/integration_tests/resources/config/spot-sample-pipeline-training/config.yaml]: done

File: components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+PipelineDefinition: ../../../../../samples/contrib/aws-samples/simple_train_pipeline/training-pipeline.py
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml:1]: delete this

File: components/aws/sagemaker/tests/integration_tests/component_tests/test_train_component.py (hunk 5 above)
Comment (surajkota) [reply components/aws/sagemaker/tests/integration_tests/component_tests/test_train_component.py:23]: remove canary marker
```
pytest.param(
            "resources/config/fsx-mnist-training",
            marks=pytest.mark.fsx_test
        ),
        "resources/config/spot-sample-pipeline-training"
```

File: components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml (hunk 7 above)
Comment (akartsky) [reply components/aws/sagemaker/tests/integration_tests/resources/config/sample-pipeline-training/config.yaml]: done

**Comment by chases2** [comment]
/hold
This PR is failing to merge due to the Travis CI test

**Comment by akartsky** [comment]
@chases2 
when will be able to merge this ?

**Pull request merged**